<?xml version="1.0"?>
<robot name="lio" xmlns:xacro="http://tixiaoshan.github.io/">
  <xacro:property name="PI" value="3.1415926535897931" />
  
  <!-- laser or base_footprint height from ground: 0.3 m -->
  <link name="lidar_link"></link>

    <link name="base_footprint"></link>
  <joint name="lidar_joint" type="fixed">
    <parent link="lidar_link"/>
    <child link="base_footprint" />
    <origin xyz="0.28 0 0" rpy="0 0 0" /> 
  </joint>

 <link name="base_link"></link>
  <!-- velodyne height from baselink: 1.43m -->
  <joint name="base_link_joint" type="fixed">
    <parent link="base_footprint"/>
    <child link="base_link" />
    <origin xyz="0 0 0.1" rpy="0 0 0" />
  </joint>
  

  <!--roughly measured: laser center 0.8 ahead: use literally hand_eye_calib-->
  <link name="laser_link"></link>
  <joint name="laser_joint" type="fixed">
    <parent link="lidar_link"/>
    <child link="laser_link" />
    <origin xyz="0.94 0 0.1" rpy="0 0 0" /> 
  </joint>

</robot>
